pkgs/http_profile/lib/http_profile.dart

```diff
@@ -2,8 +2,186 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
+import 'dart:async' show StreamController, StreamSink;
+import 'dart:developer'
+    show addHttpClientProfilingData, getHttpClientProfilingData, Timeline;
 import 'dart:io';
 
+/// Describes an event that occurred while issuing an HTTP request.
+class HttpProfileRequestEvent {
+  /// The time at which the event occurred, in microseconds.
+  final int timestamp;
+  final String name;
+  final Map<String, dynamic>? arguments;
+
+  HttpProfileRequestEvent(this.timestamp, this.name, this.arguments);
+
+  Map<String, dynamic> toJson() => <String, dynamic>{
+        'timestamp': timestamp,
+        'event': name,
+        if (arguments != null) 'arguments': arguments,
+      };
+}
+
+/// Describes proxy authentication details associated with an HTTP request.
+class HttpProfileProxyData {
+  final String? host;
+  final String? username;
+  final bool? isDirect;
+  final int? port;
+
+  HttpProfileProxyData({this.host, this.username, this.isDirect, this.port});
+
+  Map<String, dynamic> toJson() => <String, dynamic>{
+        if (host != null) 'host': host,
+        if (username != null) 'username': username,
+        if (isDirect != null) 'isDirect': isDirect,
+        if (port != null) 'port': port,
+      };
+}
+
+/// Describes details about an HTTP request.
+class HttpProfileRequestData {
+  Map<String, dynamic>? connectionInfo;
+
+  /// The content length of the request, in bytes.
+  int? contentLength;
+
+  /// The cookies presented to the server (in the 'cookie' header).
+  List<String>? cookies;
+
+  /// Events that occurred while issuing the request.
+  List<HttpProfileRequestEvent>? events;
+
+  /// The error associated with the failed request.
+  String? error;
+
+  /// Whether redirects were followed automatically.
+  bool? followRedirects;
+
+  Map<String, dynamic>? headers;
```

Comment:
At some point you'll have to document the semantics of fields like this - what can the the `dynamic` part be? Is it String|List<String>?

Reply:
I _think_ header values can only be `String`, but I could be wrong.

Reply:
Done.